### PR TITLE
himportscan upgrade for GHC 9.6

### PR DESCRIPTION
**Pull request by wavewave**
~this is on top of #59.~ rebased.

### CODE REVIEW

**Comment by avdv** [comment]
@mergifyio rebase

File: himportscan/src/HImportScan/GHC/FakeSettings9_6.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      , platform_constants = Nothing
+      , platformHasLibm = True
+      }
+
+fakeLlvmConfig :: LlvmConfig
+fakeLlvmConfig = LlvmConfig [] []
+
+#else
+
+module HImportScan.GHC.FakeSettings9_4 where
```
Comment (avdv) [inline himportscan/src/HImportScan/GHC/FakeSettings9_6.hs:65]: This should be `HImportScan.GHC.FakeSettings9_6`

**Comment by avdv** [comment]
Thank you! Could you also add an entry to the CircleCI job matrix for GHC 9.6 please?

File: himportscan/src/HImportScan/GHC/FakeSettings9_6.hs (hunk 1 above)
Comment (wavewave) [reply himportscan/src/HImportScan/GHC/FakeSettings9_6.hs]: Now updated. Thanks for the catch!

**Comment by avdv** [comment]
@wavewave  Oh, you also need to add a stack config for 9.6.2. And a lock file for it. Do you know how or should I add this?

**Comment by wavewave** [comment]
Ah. thanks. Just saw your message here. actually, I don't know how to add that snapshot-*.json file. Could you inform me the command? thanks!

**Comment by wavewave** [comment]
now i read `rules_haskell` documentation and seems like 
```
bazel run @stackage-unpinned//:pin
```
was the command to generate. :-)

**Comment by wavewave** [comment]
Ok. the missing snapshot file problem got solved but now I have this.. 🤔 
```
ERROR: /home/runner/.cache/bazel/_bazel_runner/2f1e0776f91a8e59682a44ab965a403c/external/stackage/BUILD.bazel:623:26: in haskell_toolchain_library rule @stackage//:template-haskell: 
Traceback (most recent call last):
	File "/home/runner/.cache/bazel/_bazel_runner/2f1e0776f91a8e59682a44ab965a403c/external/rules_haskell/haskell/private/haskell_impl.bzl", line 805, column 13, in haskell_toolchain_library_impl
		fail(
Error in fail: 
template-haskell is not a toolchain library.
Check that it ships with your version of GHC.
The following toolchain libraries are available:
{}
```

**Comment by avdv** [comment]
> Ok. the missing snapshot file problem got solved but now I have this.. 🤔
> 
> ```
> ERROR: /home/runner/.cache/bazel/_bazel_runner/2f1e0776f91a8e59682a44ab965a403c/external/stackage/BUILD.bazel:623:26: in haskell_toolchain_library rule @stackage//:template-haskell: 
> Traceback (most recent call last):
> 	File "/home/runner/.cache/bazel/_bazel_runner/2f1e0776f91a8e59682a44ab965a403c/external/rules_haskell/haskell/private/haskell_impl.bzl", line 805, column 13, in haskell_toolchain_library_impl
> 		fail(
> Error in fail: 
> template-haskell is not a toolchain library.
> Check that it ships with your version of GHC.
> The following toolchain libraries are available:
> {}
> ```

This is https://github.com/tweag/rules_haskell/issues/1920 which has been fixed in rules_haskell 0.18.

**Comment by wavewave** [comment]
Thanks! updated rules_haskell to 0.18.

**Comment by wavewave** [comment]
Certainly Cabal specified in the snapshot (tweag/cabal `4f8c4ab`) is not compatible with GHC 9.6. I am not familiar with the modification needed (first of all, why should we use a custom version of Cabal?). Would you check whether there is a working version (or whether needed for a special Cabal)? BTW, on my project, it was working with stock Cabal on GHC 9.6.

**Comment by avdv** [comment]
> Certainly Cabal specified in the snapshot (tweag/cabal `4f8c4ab`) is not compatible with GHC 9.6. I am not familiar with the modification needed (first of all, why should we use a custom version of Cabal?). Would you check whether there is a working version (or whether needed for a special Cabal)? BTW, on my project, it was working with stock Cabal on GHC 9.6.

This is related to a problem in Cabal (see https://github.com/tweag/rules_haskell/issues/1832) which has been fixed in Cabal 3.10.1. GHC >= 9.6 includes a recent enough version of Cabal so we don't need the workaround anymore.

**Pull request closed without merging**